### PR TITLE
Update registrant API and QR code generation

### PR DESCRIPTION
**Pull request by nop33**

### CODE REVIEW

File: indico/MaKaC/services/implementation/registration.py [hunk 1]
```diff
@@ -101,8 +102,9 @@ def _getAnswer(self):
                    "date": format_date(self._conf.getAdjustedStartDate()),
                    "server": {"baseUrl": baseURL,
                               "consumerKey": checkin_app.client_id,
-                              "consumerSecret": checkin_app.client_secret,
-                              }
+                              "auth_url": url_for('oauth.oauth_authorize'),
+                              "token_url": url_for('oauth.oauth_token')
+                              },
```
Comment (ThiefMaster) [inline indico/MaKaC/services/implementation/registration.py]: broken indentation (when aligning keys, the closing `}` and if used the `,` right after it go in the previous line)

File: indico/modules/events/registration/api.py [hunk 2]
```diff
@@ -76,6 +79,34 @@ def _process_PATCH(self):
...
+    def _checkParams(self):
+        event = ConferenceHolder().getById(request.view_args['event_id'], True)
+        if event is None:
+            raise NotFound("No such event")
+        if not event.canManageRegistration(request.oauth.user):
+            raise Forbidden()
+        self.event = event
+
+    def _process_GET(self):
+        registrants = self.event.getRegistrantsList()
```
Comment (ThiefMaster) [inline indico/modules/events/registration/api.py]: why assign it and not iterate over it directly?

File: indico/modules/events/registration/api.py [hunk 3]
```diff
@@ -76,6 +79,34 @@ def _process_PATCH(self):
...
+        registrant_list = []
```
Comment (ThiefMaster) [inline indico/modules/events/registration/api.py]: rename to `registrants` afterwards

File: indico/modules/events/registration/api.py [hunk 4]
```diff
@@ -76,6 +79,34 @@ def _process_PATCH(self):
...
+        for registrant in registrants:
+            registrant_entry = {
+                "registrant_id": registrant.getId(),
+                "checked_in": registrant.isCheckedIn(),
+                "full_name": registrant.getFullName(title=True, firstNameFirst=True),
+                "checkin_secret": registrant.getCheckInUUID(),
+            }
+            regForm = self.event.getRegistrationForm()
```
Comment (ThiefMaster) [inline indico/modules/events/registration/api.py]: why not move this up and add `personal_data` to the `registrant_entry = {...}` dict literal?

File: indico/MaKaC/services/implementation/registration.py (hunk 1 above)
Comment (nop33) [inline indico/MaKaC/services/implementation/registration.py]: :+1:

File: indico/modules/events/registration/api.py (hunk 2 above)
Comment (nop33) [inline indico/modules/events/registration/api.py]: Good point. Will fix it :+1:

File: indico/modules/events/registration/api.py (hunk 3 above)
Comment (nop33) [inline indico/modules/events/registration/api.py]: :+1:

File: indico/modules/events/registration/api.py (hunk 4 above)
Comment (nop33) [inline indico/modules/events/registration/api.py]: :+1:

File: indico/MaKaC/services/implementation/registration.py [hunk 5]
```diff
@@ -101,8 +102,8 @@ def _getAnswer(self):
                    "date": format_date(self._conf.getAdjustedStartDate()),
                    "server": {"baseUrl": baseURL,
                               "consumerKey": checkin_app.client_id,
-                              "consumerSecret": checkin_app.client_secret,
-                              }
+                              "auth_url": url_for('oauth.oauth_authorize'),
+                              "token_url": url_for('oauth.oauth_token')}
                    }
```
Comment (ThiefMaster) [inline indico/MaKaC/services/implementation/registration.py]: Don't you want `_external=True` for those URLs so you get absolute ones?

Also, the `}` should go in the previous line...

Comment (nop33) [inline indico/MaKaC/services/implementation/registration.py]: Right, I didn't think about that. Will do.

**Comment by ThiefMaster** [comment]
Merged 1517026

**Pull request closed without merging**